text-serializer-gson-legacy-impl/src/main/java/net/kyori/adventure/text/serializer/gson/legacyimpl/NBTLegacyHoverEventSerializer.java

```diff
@@ -31,14 +31,17 @@
  * A legacy {@link HoverEvent} serializer.
  *
  * @since 4.3.0
+ * @deprecated For removal, use {@link net.kyori.adventure.text.serializer.json.legacyimpl.NBTLegacyHoverEventSerializer}
  */
-public interface NBTLegacyHoverEventSerializer extends LegacyHoverEventSerializer {
+@Deprecated
```

Comment:
If you're deprecating stuff for removal you can use `@ApiStatus.ScheduledForRemoval`. This applies throughout.